Mono.Debugging.Soft/PortablePdbData.cs

```diff
@@ -77,73 +80,30 @@ internal class SoftScope
 			public int LiveRangeEnd;
 		}
 
-		class JsonSourceLink
+		Stream GetStream()
 		{
-			[JsonProperty ("documents")]
-			public Dictionary<string, string> Maps { get; set; }
+			if (pdbBytes != null)
+				return new MemoryStream (pdbBytes);
+			return new FileStream (pdbFileName, FileMode.Open, FileAccess.Read, FileShare.Read);
 		}
 
-		class SourceLinkMap
+		public string GetSourceLinkBlob ()
 		{
-			public string RelativePathWildcard { get; }
-			public string UriWildcard { get; }
-
-			public SourceLinkMap (string relativePathWildcard, string uriWildcard)
-			{
-				UriWildcard = uriWildcard;
-				RelativePathWildcard = relativePathWildcard;
-			}
-		}
+			using (var stream = GetStream()) {
+				using (var provider = MetadataReaderProvider.FromPortablePdbStream (stream)) {
+					var pdbReader = provider.GetMetadataReader ();
 
-		public SourceLink GetSourceLink (string originalFileName)
-		{
-			if (originalFileName == null)
-				return null;
-
-			foreach (var map in sourceLinkMaps.Value) {
-				var pattern = map.RelativePathWildcard.Replace ("*", "").Replace ('\\', '/');
-
-				if (originalFileName.StartsWith (pattern, StringComparison.Ordinal)) {
-					var localPath = originalFileName.Replace (pattern.Replace (".*", ""), "");
-					var httpBasePath = map.UriWildcard.Replace ("*", "");
-					// org/project-name/git-sha (usually)
-					var pathAndQuery = new Uri (httpBasePath).GetComponents (UriComponents.Path, UriFormat.SafeUnescaped).Substring (1);
-					// org/projectname/git-sha/path/to/file.cs
-					var relativePath = Path.Combine (pathAndQuery, localPath);
-					// Replace something like "f:/build/*" with "https://raw.githubusercontent.com/org/projectname/git-sha/*"
-					var httpPath = Regex.Replace (originalFileName, pattern, httpBasePath);
-					return new SourceLink (httpPath, relativePath);
-				}
+					var jsonBlob =
+						(from cdiHandle in pdbReader.GetCustomDebugInformation (EntityHandle.ModuleDefinition)
```

Comment:
Can we avoid LINQ? I think LINQ extensions make it easier to reason about captures.